### DISCUSSION TITLE
is there any way to reset touched on a specific field, not the whole form?

### DISCUSSION THREAD

**Discussion by phyolim** [react-hook-form/react-hook-form#3381] [Q&A]
To clarify. I was trying to reset only a specific field in form and set touched as false on that field.

**Comment by bluebill1049** [comment]
you can reset({ ...getValues() }, { omitState })

https://react-hook-form.com/api#reset

**Reply by phyolim** [reply]
If I do this, it reset the touched value on all fields, Is it possible to reset touched for only a single field in form?

**Reply by bluebill1049** [reply]
you have to write your own logic to only reset whatever its touched.

**Reply by phyolim** [reply]
I don't know what you mean. Would I still use the reset function? or write something myself entirely ?

EDIT: I used control.formStateRef and mutated the touched value there. and do forceRender. I'm not sure if this is the proper way.

**Reply by bluebill1049** [reply]
you still use `reset(// your logic here to produce the form value)`

**Reply by phyolim** [reply]
I still don't get it. I forked reset example here -
https://codesandbox.io/s/react-hook-form-reset-form-forked-cfcm2?file=/src/index.js
I'm trying to reset first name to "Bob" while keeping last name field as is. and I wanted to reset the touched value on firstName field only.
reset function expect an initial state object to be passed in right? is it possible to pass in values for touched in form in there too?

**Reply by ansgarsteinkamp** [reply]
Is there a solution for the problem by now? I also don't understand what bluebill1049 is saying.

**Reply by ShakurOo** [reply]
+1

**Reply by maurer2** [reply]
Hello, I think one can either use the `resetField` or `setValue` methods that are returned from the `useForm` hook. `resetField` updates the default value of a field, while `setValue` does not. It was also not available back in 2020, when this question was first asked.

Example `resetField`
`resetField('firstName, { keepDirty: false, defaultValue: 'Bob' })`

Example `setValue`
`setValue('firstName', 'Bob', { shouldDirty: false });`

Cheers
